src/bootstrap/bootstrap.py

```diff
@@ -529,6 +534,32 @@ def build_triple(self):
 
         return "{}-{}".format(cputype, ostype)
 
+    def update_submodules(self):
+        if (not os.path.isdir(os.path.join(self.rust_root, ".git"))) or \
```

Comment:
It's not valid to assume `.git` is a directory - if the rust repo itself is a git submodule and you're using particular versions of git, `.git` will be a file containing a single line (the path pointing to the *actual* `.git` directory).